Marlin/src/feature/touch/xpt2046.h

```diff
@@ -42,8 +42,11 @@ class XPT2046 {
 public:
   static void init(void);
   static uint8_t read_buttons();
-private:
+  bool getTouchPoint(uint16_t &x, uint16_t &y);
   static bool isTouched();
+  inline void waitForRelease(void) { while (isTouched()) { /* nada */ } }
+  inline void waitForTouch(uint16_t &x, uint16_t &y) { while (!getTouchPoint(x, y)) { /* nada */ } }
```

Comment:
If you call either of these the watchdog will reset the board. They must call `idle` within these loops.

Reply:
hmm i see your point, but doesnt actually do that in the calibration applet... maybe because it should be done with all steppers and heaters off... so not while printing

Reply:
not sure what the code do, but the end results that the users find  in the main branch of the Marlin 2.xx bug branch is "wrongly" assigned. (configuration.h)

I thing most of people that re reporting their touch is not working are due to swapped values for the Xcalibration Ycalibration and x off set and y offset.
 I have the MKS-Robin V2.1 with TFT touch 1.1 and the following value works for me
#define XPT2046_X_CALIBRATION   -11876
  #define XPT2046_Y_CALIBRATION  8755
  #define XPT2046_X_OFFSET       346
  #define XPT2046_Y_OFFSET        -20

the value are made from the calibration tool from jmz52. under issue #9771
I would suggest if not yet done to add a feature of LCD flip (touch portion). (I know this is not the right place and way of suggesting, but just let out my thoughts here, hope you won't mind)

many thanks for your help and time put into this FW and supporting this great community!!!

Reply:
see how we made that for the U20 in the samples : https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/config/examples/Alfawise/U20/Configuration.h#L2136-L2163

Users have to put a define with the TS version in the platformio.ini for now... but im not sure your robin tft versions match our 2 versions... i dont think so

Reply:
> doesnt actually do that in the calibration applet

The calibration applet doesn't enable the watchdog reset.

Reply:
no and its not able to disable it... its ran from a menu without issues... except maybe some touch accuracy actually... Sorry i was discussing a problem with the #15113 and forgot the notice to answer you... It look like the SKR pro could require the "AVR" file...